apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -122,12 +124,34 @@ on_start(InstId, Config) ->
                     client_id => ClientId
                 }
             ),
-            throw(
-                "Failed to start Kafka client. Please check the logs for errors and check"
-                " the connection parameters."
-            )
+
+            case check_if_topic_exists(Hosts, KafkaConfig, KafkaTopic) of
+                unhealthy_target ->
+                    throw(unhealthy_target);
+                _ ->
+                    throw(
+                        "Failed to start Kafka client. Please check the logs for errors and check"
+                        " the connection parameters."
+                    )
+            end
     end.
 
+check_if_topic_exists(Hosts, KafkaConfig, KafkaTopic) ->
+    % wolff starts a socket to get metadata and closes it asynchronously (calling a spawn to
```

Comment:
then setting trap_exit immediately after would not help? it could happen that the trap_exist is set back to false before the socket is down.
better solution is to span_monitor a process to do it and drain the DOWN message